code_generator/GdkCodeGenerator/Templates/UnityGameObjectCommandHandlersGenerator.tt

```diff
@@ -50,10 +78,44 @@ namespace <#= qualifiedNamespace #>
             [InjectionCondition(InjectionCondition.RequireComponentWithAuthority)]
             public class CommandRequestHandler : IInjectable
             {
+                private Entity entity;
+                private readonly EntityManager entityManager;
+                private readonly ILogDispatcher logger;
+
                 public CommandRequestHandler(Entity entity, EntityManager entityManager, ILogDispatcher logger)
                 {
+                    this.entity = entity;
+                    this.entityManager = entityManager;
+                    this.logger = logger;
+                }
 
+<# foreach(var commandDetails in commandDetailsList) {
+           var commandRequestType = commandDetails.CommandName + ".ReceivedRequest";
+           var delegateList = commandDetails.CamelCaseName + "Delegates";
+#>
+                private readonly List<Action<<#= commandRequestType #>>> <#= delegateList #> = new List<Action<<#= commandRequestType #>>>();
+                public event Action<<#= commandRequestType #>> On<#= commandDetails.CommandName #>Request
+                {
+                    add => <#= delegateList #>.Add(value);
+                    remove => <#= delegateList #>.Remove(value);
+                }
+
+                public void On<#= commandDetails.CommandName #>RequestInternal(<#= commandRequestType #> request)
```

Comment:
As it is, this is visible to the user right? We could avoid that by having a user-facing interface type that includes the events above but not this method, that might be better

Reply:
Why not just make this internal? User facing interface types aren't something I'm fond of.

Reply:
Because for some reason I forgot they are in a different assembly. That definitely makes things easier :D

Reply:
Yeah, this is an oversight on my part. I will think about a way to hide this method from users.

Reply:
I made the `On<CommandName>RequestInternal` method internal. That seems to do the trick.